Feature #28 - waiting, review - controller 작성

chanwooo opened the pull request:
WaitingController 작성

chanwooo committed:
Hotfix #34-스프링부트어드민 추가, 머지 후 실행안됨 (#35)

* #23-스프링 액추에이터 추가, 어드민 클라이언트 추가

* #23-springboot admin 버전 명시

chanwooo committed:
Merge branch 'feature-#21' into feature-#28

chanwooo committed:
Feature #21-QuingService (#24)

* #21-변수명 변경 phone -> phoneNumber

* #21-타입변경 Integer -> int

* #21-안쓰는 파일 삭제

* #21-사용하지않는 필드 제거

* #21-QuingService 일부 구현
append
getList
countForward
doneWaiting

* #21-QuingService test 일부 구현

* #21-주석 삭제

* #21-append 중복 방지 로직 추가

* #21-waiting cancel 구현

* #21-exception 패키지 이동

* #21-test코드 작성

* Feature #22-waiting - 사용자에게 메시지 send 구현 (#27)

* #21-cancel 리턴 타입 변경

* #22-message 전송 관련 부분 print로 임시 구현

* #22-message 전송 관련 부분 test 작성

* #22-test에 불필요한 공백 제거

* #22- test코드 logger 적용

* #22- test코드 given, when, then 분리

* #22- test코드 logger 적용

* #22-리뷰반영
- 테스트 공백, 주석 정리
- 테스트 이름 변경 appendExistThrowDuplicateException

* # Conflicts:
#	src/main/java/flab/quing/user/User.java
#	src/main/java/flab/quing/user/UserRepository.java

* #21-user response import

chanwooo committed:
#37-storeManager가 관리하는 store 연결 (1:1)
- response에서 password 삭제

chanwooo committed:
Merge branch 'feature-#37' into feature-#28

chanwooo committed:
#28-QuingController 작성
- 로그인한 가게(storeManager-store)의 대기열 목록(get quing)
- 로그인한 유저의 현재 대기순서(get quing/count-forward)
- 대기 생성 (post quing)
- WaitingRequest 생성자 추가(jackson에서 필요)

chanwooo committed:
#39-quingservice import 경로 수정 (#41)

- Rating OneToOne Cascade 삭제

chanwooo committed:
Merge remote-tracking branch 'origin/develop' into feature-#28

f-lab-trey commented on src/main/java/flab/quing/waiting/QuingController.java:
```diff
@@ -0,0 +1,58 @@
+    private final QuingService quingService;
+
+
+    @GetMapping("")
```
[q]
`""`이 꼭 들어가야 compile 되나요?

chanwooo replied:
경로를 통째로 지워도 잘 동작하네요~! 삭제했습니다!

f-lab-trey commented on src/main/java/flab/quing/waiting/QuingController.java:
```diff
@@ -0,0 +1,58 @@
+    @GetMapping("")
+    public List<WaitingResponse> list(
+            @SessionAttribute(name = "AUTH_STORE")
+            StoreManagerResponse storeManagerResponse
```
[minor]
`private`이 붙으면 좋겠네요.

chanwooo replied:
controller method에 접근지시자가 실제 동작에는 별 의미가 없다는 글을 봤는데
그렇다면 다른 경로에서 해당 메소드에 접근할수 없도록 private를 추천한다는걸로 이해를 해도 좋을까요?

chanwooo replied:
private으로 하니까 테스트작성이 안되어서 default로 변경하였는데 어떤게 좋을지 잘 모르겠네요
https://github.com/f-lab-edu/quing/pull/40/commits/b09ebdd32edaa83383b90105d6728eb858caa2e9

f-lab-trey replied:
Method에 붙이면 호출이 안되지 않나요? 모든 parameter에 붙여달라는 의미였습니다. Method는 모두 public으로 해도 상관없을 거 같습니다.

f-lab-trey commented on src/main/java/flab/quing/waiting/QuingController.java:
```diff
@@ -0,0 +1,58 @@
+
+    @GetMapping("count-forward")
+    public long countForward(@SessionAttribute(name = "AUTH_USER")
+                             UserResponse userResponse
```
ditto

f-lab-trey commented on src/main/java/flab/quing/waiting/QuingController.java:
```diff
@@ -0,0 +1,58 @@
+    public WaitingResponse append(
+            @SessionAttribute(name = "AUTH_USER")
+            UserResponse userResponse,
+            @RequestBody long storeId
```
`@RequestBody`로 `WaitingRequest`가 와야하지 않을까요? 현재 상태면, caller가 JSON 형태의 body를 어떤 형태로 서버로 전송해야되나요?

chanwooo replied:
로그인이 되어있다면 userid가 채워지니 storeid만 입력받으면 된다고 생각했는데
둘다 받고 입력값을 비교하거나 storeid를 받기위한 request를 만들어야겠네요!

chanwooo committed:
#28-리뷰반영
- 기본경로 의미없는 ("") 삭제
- method private로 변경

chanwooo committed:
#28-Waiting done, cancel 추가

chanwooo committed:
#28-getByuser 추가
- done, cancel을 하기위해 userid로 waiting을 가져오는게 필요해서 추가

chanwooo committed:
#28-append request 추가

chanwooo committed:
#28-storeId로 변경

chanwooo committed:
#28-QuingController test 작성

chanwooo committed:
#28-private 제거
- controller는 다른데서 호출될일이 없겠거니 하고 private를 달았는데 이것 때문에 테스트가 불가하여 Access Modifier를 defualt로 변경..

chanwooo commented:
review controller작성은 다른 이슈로 분리하려고해요.
일단 quing controller 완료되었으니 pr로 변경하겠습니다.
